### PR TITLE
Vschema endtoend GO testcases

### PR DESCRIPTION
**Pull request by ajeetj**
Vschema endtoend testcases in GO migrated from Python ([reference](https://github.com/planetscale/vitess/blob/master/test/vschema.py))

Do not merge as I have one question in this PR.

### CODE REVIEW

File: go/vt/vtgate/endtoend/vschema/vschema_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	exec(t, conn, "begin")
+	exec(t, conn, "ALTER VSCHEMA ADD TABLE main")
+	exec(t, conn, "ALTER VSCHEMA ADD TABLE vt_user")
+	exec(t, conn, "commit")
+
+	//Test Showing Tables
+	qr = exec(t, conn, "SHOW VSCHEMA TABLES")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARCHAR("dual")] [VARCHAR("main")] [VARCHAR("vt_user")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
```
Comment (ajeetj) [inline go/vt/vtgate/endtoend/vschema/vschema_test.go]: @deepthi above test case fails intermittently and I am not able to figure out the reason for that. 
Sometimes none of the requested Vschema gets created and sometimes only one of them is present. 
Here is the screenshot when it fails. 

Screen Shot 2019-10-18 at 1 53 20 PM

Comment (deepthi) [reply go/vt/vtgate/endtoend/vschema/vschema_test.go]: @sougou is there some caching that needs to be considered?

File: go/vt/vtgate/endtoend/vschema/vschema_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	val varchar(128),
+	primary key(id)
+) Engine=InnoDB;
+`
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitCode := func() int {
```
Comment (sougou) [inline go/vt/vtgate/endtoend/vschema/vschema_test.go]: You probably copied my bad code :).
The better pattern is to have this function return an error, and then do this after the call:
```
if err != nil {
  fmt.Printf("%v\n", err)
  os.Exit(1)
}
```

File: go/vt/vtgate/endtoend/vschema/vschema_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// Test the blank database with no vschema
+	exec(t, conn, "insert into vt_user (id,name) values(1,'test1'), (2,'test2'), (3,'test3'), (4,'test4')")
+	qr := exec(t, conn, "select id, name from vt_user order by id")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(1) VARCHAR("test1")] [INT64(2) VARCHAR("test2")] [INT64(3) VARCHAR("test3")] [INT64(4) VARCHAR("test4")]]`; got != want {
```
Comment (sougou) [inline go/vt/vtgate/endtoend/vschema/vschema_test.go]: We're switching to using the `assert` package which avoids boiler-plating. Here's an example: https://github.com/vitessio/vitess/blob/master/go/vt/sqlparser/like_filter_test.go#L29

File: go/vt/vtgate/endtoend/vschema/vschema_test.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		if err := cfg.InitSchemas("ks", schema, nil); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.RemoveAll(cfg.SchemaDir)
+			return 1
+		}
+		defer os.RemoveAll(cfg.SchemaDir)
+
+		cfg.TabletHostName = *tabletHostName
+
+		//List of users authorized to execute vschema ddl operations
```
Comment (sougou) [inline go/vt/vtgate/endtoend/vschema/vschema_test.go]: Need space between `//` and statement. Same comments below.

File: go/vt/vtgate/endtoend/vschema/vschema_test.go (hunk 1 above)
Comment (sougou) [reply go/vt/vtgate/endtoend/vschema/vschema_test.go]: I think this has to do with the delay of the vtcombo watching the SrvVSchema. You could put a sleep, but that may not completely eliminate flakyness.

You can instead check this in a loop for 10-20 seconds, and then fail.

Comment (ajeetj) [reply go/vt/vtgate/endtoend/vschema/vschema_test.go]: I tried looping and waiting for 60 seconds but that still not guaranteed for updated vschema. Then I tried to add a select statement after the vschema updated and looks like that solved it. The test cases are not failing. 

Can you please review again if that approach is fine?

File: go/vt/vtgate/endtoend/vschema/vschema_test.go (hunk 2 above)
Comment (ajeetj) [reply go/vt/vtgate/endtoend/vschema/vschema_test.go]: Updated code with suggestions

File: go/vt/vtgate/endtoend/vschema/vschema_test.go (hunk 3 above)
Comment (ajeetj) [reply go/vt/vtgate/endtoend/vschema/vschema_test.go]: Updated test to use `assert`

**Comment by ajeetj** [comment]
Closing this PR as I have moved this file in the new go cluster env. 
Related PR https://github.com/planetscale/vitess/pull/17

**Pull request closed without merging**